bigchaindb/common/transaction.py

```diff
@@ -592,51 +585,31 @@ def __init__(self, operation, asset, fulfillments=None, conditions=None,
                 version (int): Defines the version number of a Transaction.
 
         """
-        if version is not None:
-            self.version = version
-        else:
-            self.version = self.__class__.VERSION
-
-        if timestamp is not None:
-            self.timestamp = timestamp
-        else:
-            self.timestamp = gen_timestamp()
-
         if operation not in Transaction.ALLOWED_OPERATIONS:
             allowed_ops = ', '.join(self.__class__.ALLOWED_OPERATIONS)
             raise ValueError('`operation` must be one of {}'
                              .format(allowed_ops))
-        else:
-            self.operation = operation
 
-        # If an asset is not defined in a `CREATE` transaction, create a
-        # default one.
-        if asset is None and operation == Transaction.CREATE:
-            asset = Asset()
-
-        if not isinstance(asset, Asset):
+        # Only assets for 'CREATE' operations can be un-defined.
+        if operation != Transaction.CREATE and not isinstance(asset, Asset):
             raise TypeError('`asset` must be an Asset instance')
```

Comment:
I think this part of the translation isn't right. An `Asset` should only be automatically created if it's a `CREATE` and no asset was given.